Change description of underscore-only variables

henriquebecker91 opened the pull request:
Spawned by https://discourse.julialang.org/t/class-of-variables/83892

The rvalue/lvalue description does not seem benefitial. It confuse programmers that are not programming language nerds (it is a cool concept to learn, but this does not seem the place to learn it), and even programming language nerds may object. I for example, find kinda of a stretch to call `___` a lvalue because the linked wikipedia page says:

> An l-value refers to an object that persists beyond a single expression. An r-value is a temporary value that does not persist beyond the expression that uses it.[3]

Considering this description, the `___` matches more a rvalue than an lvalue (even if it is semantically a location and appear always in the left-hand side of an assignment).

henriquebecker91 committed:
Change description of underscore-only variables

Spawned by https://discourse.julialang.org/t/class-of-variables/83892

The rvalue/lvalue description does not seem benefitial. It confuse programmers that are not programming language nerds (it is a cool concept to learn, but this does not seem the place to learn it), and even programming language nerds may object. I for example, find kinda of a stretch to call `___` a lvalue because the linked wikipedia page says:

> An l-value refers to an object that persists beyond a single expression. An r-value is a temporary value that does not persist beyond the expression that uses it.[3]

Considering this description, the `___` matches more a rvalue than an lvalue (even if it is semantically a location and appear always in the left-hand side of an assignment).

KristofferC commented:
The error message refers to them as R-values though so as long that is the case, doesn't it make sense to also have that in the documentation?

henriquebecker91 commented:
Yes, I forgot to account the error message, but it just says they are **not** `rvalues`. I will change the text to at least address this point, but I find that we do not need to go defining them as `lvalues` as done before (as they do not retain a real location).

henriquebecker91 committed:
Taken @KristofferC suggestions into consideration 

Added back a mention to `___` not being `rvalue`s (but nothing about them being `lvalue`s as they do not keep value nor are a real persistent location).
